### PR TITLE
DROID-1460 Editor | Enhancement | Use-current-object-as-template action

### PR DESCRIPTION
**Pull request by konstantiniiv**

### CODE REVIEW

File: app/src/main/java/com/anytypeio/anytype/ui/editor/EditorFragment.kt [hunk 1]
```diff
@@ -2088,6 +2094,30 @@ open class EditorFragment : NavigationFragment<FragmentEditorBinding>(R.layout.f
         }
     }
 
+    open fun observeSelectingTemplate() {
```
Comment (konstantiniiv) [inline app/src/main/java/com/anytypeio/anytype/ui/editor/EditorFragment.kt:2097]: https://developer.android.com/guide/navigation/use-graph/programmatic#additional_considerations

File: app/src/main/java/com/anytypeio/anytype/ui/sets/ObjectSetFragment.kt [hunk 2]
```diff
@@ -1166,13 +1167,25 @@ open class ObjectSetFragment :
     )
 
     private fun observeSelectingTemplate() {
-        findNavController().currentBackStackEntry?.savedStateHandle?.getLiveData<String>(
-            ARG_TEMPLATE_ID,
-            SELECTED_TEMPLATE_INITIAL_VALUE
-        )?.observe(viewLifecycleOwner) { template: Id ->
-            Timber.d("Get result from EditorTemplateFragment: $template")
-            if (template.isNotEmpty()) vm.proceedWithCreatingNewDataViewObject(template)
+        val navController = findNavController()
```
Comment (konstantiniiv) [inline app/src/main/java/com/anytypeio/anytype/ui/sets/ObjectSetFragment.kt:1170]: https://developer.android.com/guide/navigation/use-graph/programmatic#additional_considerations

File: app/src/main/java/com/anytypeio/anytype/ui/templates/EditorTemplateFragment.kt [hunk 3]
```diff
@@ -73,13 +73,14 @@ class EditorTemplateFragment : EditorFragment() {
         }
     }
 
+    override fun observeSelectingTemplate() {}
```
Comment (uburoiubu) [inline app/src/main/java/com/anytypeio/anytype/ui/templates/EditorTemplateFragment.kt:76]: `please add comment "// Do nothing"`

File: presentation/src/main/java/com/anytypeio/anytype/presentation/objects/menu/ObjectMenuViewModel.kt [hunk 4]
```diff
@@ -229,15 +245,47 @@ class ObjectMenuViewModel(
...
+                },
+                onFailure = {
+                    Timber.e(it, "Error while creating template from object")
+                    _toasts.emit(SOMETHING_WENT_WRONG_MSG)
+                }
+            )
+        }
+    }
+
+    private fun createOpenTemplateCommand(ctx: Id, template: Id): Command.OpenTemplate {
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/objects/menu/ObjectMenuViewModel.kt:278]: `buildOpenTemplateCommand`

**Review by uburoiubu — APPROVED** [review]

File: presentation/src/main/java/com/anytypeio/anytype/presentation/objects/menu/ObjectMenuViewModel.kt (hunk 4 above)
Comment (konstantiniiv) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/objects/menu/ObjectMenuViewModel.kt:278]: done

File: app/src/main/java/com/anytypeio/anytype/ui/templates/EditorTemplateFragment.kt (hunk 3 above)
Comment (konstantiniiv) [reply app/src/main/java/com/anytypeio/anytype/ui/templates/EditorTemplateFragment.kt:76]: done

**Pull request merged**